pulsar-client-cpp/lib/PeriodicTask.cc

```diff
@@ -27,9 +27,14 @@ void PeriodicTask::start() {
     }
     state_ = Ready;
     if (periodMs_ >= 0) {
-        auto self = shared_from_this();
+        std::weak_ptr<PeriodicTask> weakSelf{shared_from_this()};
         timer_.expires_from_now(boost::posix_time::millisec(periodMs_));
-        timer_.async_wait([this, self](const ErrorCode& ec) { handleTimeout(ec); });
+        timer_.async_wait([weakSelf](const ErrorCode& ec) {
```

Comment:
Is there a way to use the unit test to cover the destructor that needs to be executed?

Reply:
I didn't get it why did you mention the destructor. Using `weak_ptr` here is to avoid a cyclic reference count case. Each object managed by a `shared_ptr` could cause cyclic reference. Using `weak_ptr` is a best practice for it.

Reply:
Just want to discuss whether there is a way to use unit tests to cover the cyclic references that are not allowed.